### PR TITLE
migrate: stop deleting _etcd

### PR DESCRIPTION
**Pull request by barakmich**

### CODE REVIEW

File: migrate/snapshot.go [hunk 1]
```diff
@@ -157,10 +180,11 @@ func mangleRoot(n *node) *node {
 	}
 	newRoot.Children["1"] = n
 	etcd := n.Children["_etcd"]
-	delete(n.Children, "_etcd")
+	newEtcd := deepCopyNode(etcd, newRoot)
```
Comment (yichengq) [inline migrate/snapshot.go]: what if we just `deepCopyNode(etcd.Children["machines"])`?
i think it is good as long as jed can migrate discovery record.

**Comment by yichengq** [comment]
@jedsmith Could you try to migrate it with this patch? You should be able to get a lot of things from `etcdctl ls --recursive /_etcd` if it succeeds.

File: migrate/snapshot.go [hunk 2]
```diff
@@ -157,10 +180,11 @@ func mangleRoot(n *node) *node {
...
 	replacePathNames(n, "/", "/1/")
-	fixEtcd(etcd)
-	newRoot.Children["0"] = etcd
+	fixEtcd(newEtcd)
+	newRoot.Children["0"] = newEtcd
+	newRoot.Children["0"].Parent = newRoot
```
Comment (yichengq) [inline migrate/snapshot.go]: do we need this line? i think the parent has been set in deepCopyNode.

Comment (barakmich) [inline migrate/snapshot.go]: correct, this is unnecessary

File: migrate/snapshot.go (hunk 1 above)
Comment (barakmich) [inline migrate/snapshot.go]: that would normally work, but it'd also involve changing fixEtcd (which could be done) to manually create the new node.

Probably should be done that way too. I just wanted to get the PR out there first.

**Comment by yichengq** [comment]
@barakmich I have tested it with discovery service data dir, and it can work and `/_etcd/registry/` is migrated.
lgtm after addressing the comments

**Comment by yichengq** [comment]
lgtm

**Pull request merged**